Tests/PubNubTests/Networking/Routers/PresenceRouterTests.swift

```diff
@@ -357,3 +377,133 @@ final class PresenceEndpointTests: XCTestCase {
     wait(for: [expectation], timeout: 1.0)
   }
 }
+
+// MARK: - Heartbeat Tests
+
+extension PresenceRouterTests {
+  func testHeartbeat_Router() {
+    let router = PresenceRouter(.heartbeat(channels: [channelName], groups: [], presenceTimeout: nil), configuration: config)
+
+    XCTAssertEqual(router.endpoint.description, "Heartbeat")
+    XCTAssertEqual(router.category, "Heartbeat")
+    XCTAssertEqual(router.service, .presence)
+  }
+
+  func testHeartbeat_Router_ValidationError() {
+    let router = PresenceRouter(.heartbeat(channels: [], groups: [], presenceTimeout: nil), configuration: config)
+
+    XCTAssertEqual(router.validationError?.pubNubError?.details.first,
+                      ErrorDescription.missingChannelsAnyGroups)
+  }
+
+  func testHeartbeat_Router_Channels() {
+    let router = PresenceRouter(.heartbeat(channels: [channelName], groups: [], presenceTimeout: nil), configuration: config)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line should be 120 characters or less: currently 125 characters](https://app.codacy.com/app/PubNub/swift/pullRequest?prid=4513555)